ignition_ros2_control/src/ignition_system.cpp

```diff
@@ -0,0 +1,402 @@
+      }
+    }
+
+    RCLCPP_INFO_STREAM(
+      this->nh_->get_logger(), "\tState:");
+    // register the state handles
+    for (unsigned int i = 0; i < hardware_info.joints[j].state_interfaces.size(); i++) {
+      if (hardware_info.joints[j].state_interfaces[i].name == "position") {
+        RCLCPP_INFO_STREAM(this->nh_->get_logger(), "\t\t position");
+        this->dataPtr->joint_pos_state_[j] = std::make_shared<hardware_interface::StateInterface>(
+          joint_name, hardware_interface::HW_IF_POSITION, &this->dataPtr->joint_position_[j]);
+      }
+      if (hardware_info.joints[j].state_interfaces[i].name == "velocity") {
+        RCLCPP_INFO_STREAM(this->nh_->get_logger(), "\t\t velocity");
+        this->dataPtr->joint_vel_state_[j] = std::make_shared<hardware_interface::StateInterface>(
+          joint_name, hardware_interface::HW_IF_VELOCITY, &this->dataPtr->joint_velocity_[j]);
+      }
+      if (hardware_info.joints[j].state_interfaces[i].name == "effort") {
+        RCLCPP_INFO_STREAM(this->nh_->get_logger(), "\t\t effort");
+        this->dataPtr->joint_eff_state_[j] = std::make_shared<hardware_interface::StateInterface>(
+          joint_name, hardware_interface::HW_IF_EFFORT, &this->dataPtr->joint_effort_[j]);
+      }
+    }
+  }
+  return true;
+}
+
+hardware_interface::return_type
+IgnitionSystem::configure(const hardware_interface::HardwareInfo & actuator_info)
+{
+  if (configure_default(actuator_info) != hardware_interface::return_type::OK) {
+    return hardware_interface::return_type::ERROR;
+  }
+  return hardware_interface::return_type::OK;
+}
+
+std::vector<hardware_interface::StateInterface>
+IgnitionSystem::export_state_interfaces()
+{
+  std::vector<hardware_interface::StateInterface> state_interfaces;
+
+  for (unsigned int i = 0; i < this->dataPtr->joint_names_.size(); i++) {
+    state_interfaces.emplace_back(
+      hardware_interface::StateInterface(
+        this->dataPtr->joint_names_[i],
+        hardware_interface::HW_IF_POSITION,
+        &this->dataPtr->joint_position_[i]));
+  }
+  for (unsigned int i = 0; i < this->dataPtr->joint_names_.size(); i++) {
+    state_interfaces.emplace_back(
+      hardware_interface::StateInterface(
+        this->dataPtr->joint_names_[i],
+        hardware_interface::HW_IF_VELOCITY,
+        &this->dataPtr->joint_velocity_[i]));
+  }
+  for (unsigned int i = 0; i < this->dataPtr->joint_names_.size(); i++) {
+    state_interfaces.emplace_back(
+      hardware_interface::StateInterface(
+        this->dataPtr->joint_names_[i],
+        hardware_interface::HW_IF_EFFORT,
+        &this->dataPtr->joint_effort_[i]));
+  }
+  return state_interfaces;
+}
+
+std::vector<hardware_interface::CommandInterface>
+IgnitionSystem::export_command_interfaces()
+{
+  std::vector<hardware_interface::CommandInterface> command_interfaces;
+
+  for (unsigned int i = 0; i < this->dataPtr->joint_names_.size(); i++) {
```

Comment:
```suggestion
  for (unsigned int i = 0; i < this->dataPtr->joint_names_.size(); ++i) {
```

Nit: this is slightly more efficient. The same comment is valid to all other `++`